test/e2e/tests/test_notebook_instance_lifecycle.py

```diff
@@ -0,0 +1,105 @@
+# License is located at
+#
+# 	 http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+"""Integration tests for the Notebook Lifecycle configuration
+"""
+
+import pytest
+import logging
+import botocore
+
+from acktest.k8s import resource as k8s
+from acktest.resources import random_suffix_name
+from e2e import (
+    service_marker,
+    wait_for_status,
+    create_sagemaker_resource,
+    sagemaker_client,
+)
+
+from e2e.bootstrap_resources import get_bootstrap_resources
+import random
+
+from e2e.replacement_values import REPLACEMENT_VALUES
+from time import sleep
+
+
+@pytest.fixture(scope="function")
+def notebook_instance_lifecycleConfig():
+    notebook_instance_lfc_name = random_suffix_name("nblf", 32)
+    replacements = REPLACEMENT_VALUES.copy()
+    replacements["NOTEBOOK_INSTANCE_LFC_NAME"] = notebook_instance_lfc_name
+    reference, spec, resource = create_sagemaker_resource(
+        resource_plural="notebookinstancelifecycleconfigs",
+        resource_name=notebook_instance_lfc_name,
+        spec_file="notebook_instance_lifecycle_config",
+        replacements=replacements,
+    )
+    assert resource is not None
+    yield (reference, resource, spec)
+    if k8s.get_resource_exists(reference):
+        _, deleted = k8s.delete_custom_resource(reference, 10, 5)
+        assert deleted
+
+
+def get_notebook_instance_lifecycle_config(notebook_instance_lfc_name: str):
+    try:
+        resp = sagemaker_client().describe_notebook_instance_lifecycle_config(
+            NotebookInstanceLifecycleConfigName=notebook_instance_lfc_name
+        )
+        return resp
+    except botocore.exceptions.ClientError as error:
+        logging.error(
+            f"SageMaker could not find a Notebook Instance Lifecycle Configuration with the name {notebook_instance_lfc_name}. Error {error}"
+        )
+        return None
+
+
+@service_marker
+@pytest.mark.canary
+class TestNotebookInstanceLifecycleConfig:
+    def test_CreateUpdateDeleteNotebookLifecycleConfig(
+        self, notebook_instance_lifecycleConfig
+    ):
+        (reference, resource, spec) = notebook_instance_lifecycleConfig
+        assert k8s.get_resource_exists(reference)
+
+        # Getting the resource name
+        notebook_instance_lfc_name = resource["spec"].get(
+            "notebookInstanceLifecycleConfigName", None
+        )
+        assert notebook_instance_lfc_name is not None
+
+        # Verifying that its set correctly
+        lfc_before_update = get_notebook_instance_lifecycle_config(
+            notebook_instance_lfc_name
+        )
+        assert lfc_before_update["OnStart"][0]["Content"] == "cGlwIGluc3RhbGwgUElM"
+        spec["spec"]["onStart"] = [{"content": "cGlwIGluc3RhbGwgc2l4"}]
```

Comment:
same, mention the human-readable content in the comment and move the content to a variable which you can reuse to assert